Musician Module/MusicianModule.cs

```diff
@@ -0,0 +1,512 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Blish_HUD;
+using Blish_HUD.Controls;
+using Blish_HUD.Modules;
+using Gw2Sharp.WebApi.V2.Models;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+
+namespace Musician_Module
+{
+
+    [Export(typeof(ExternalModule))]
+    public class MusicianModule : ExternalModule
+    {
+        private Texture2D ICON = GameService.Content.GetTexture("musician_icon");
+        private const string DD_TITLE = "Title";
+        private const string DD_ARTIST = "Artist";
+        private const string DD_USER = "User";
+        private const string DD_HARP = "Harp";
+        private const string DD_FLUTE = "Flute";
+        private const string DD_LUTE = "Lute";
+        private const string DD_HORN = "Horn";
+        private const string DD_BASS = "Bass";
+        private const string DD_BELL = "Bell";
+        private const string DD_BELL2 = "Bell2";
+        private List<string> Instruments = new List<string>{
+           "Harp", "Flute", "Lute", "Horn", "Bell", "Bell2", "Bass"
+        };
+        private WindowTab MusicianTab;
+        private MusicPlayer MusicPlayer;
+        private HealthPoolButton StopButton;
+        private XmlMusicSheetReader xmlParser;
+        private List<SheetButton> displayedSheets;
+        private List<RawMusicSheet> Sheets;
+
+        /// <summary>
+        /// Ideally you should keep the constructor as is.
+        /// Use <see cref="Initialize"/> to handle initializing the module.
+        /// </summary>
+        [ImportingConstructor]
+        public Musician_Module([Import("ModuleParameters")] ModuleParameters moduleParameters) : base(moduleParameters) { /* NOOP */ }
+
+        /// <summary>
+        /// Define the settings you would like to use in your module.  Settings are persistent
+        /// between updates to both Blish HUD and your module.
+        /// </summary>
+        protected override void DefineSettings(SettingsManager settingsManager)
+        {
+        }
+
+        /// <summary>
+        /// Allows your module to perform any initialization it needs before starting to run.
+        /// Please note that Initialize is NOT asynchronous and will block Blish HUD's update
+        /// and render loop, so be sure to not do anything here that takes too long.
+        /// </summary>
+        protected override void Initialize()
+        {
+            Directory.CreateDirectory(Path.Combine(GameService.FileSrv.BasePath, "musician"));
```

Comment:
You don't have to do this anymore.  Since it's declared in your manifest.  When the module is enabled for the first time, it will automatically create the directory.  This main module class also has access to `this.DirectoriesManager` which allows you to get the paths to each of the directories you've declared.  https://github.com/blish-hud/Blish-HUD/blob/moduler-modules-rework/Blish%20HUD/Modules/Managers/DirectoriesManager.cs